src/lib.rs

```diff
@@ -29,12 +29,15 @@
 //! This crate uses the same syntax for format strings as the [strftime()]
 //! (http://man7.org/linux/man-pages/man3/strftime.3.html) function from the C
 //! standard library.
-
 #![doc(html_logo_url = "https://www.rust-lang.org/logos/rust-logo-128x128-blk-v2.png",
        html_favicon_url = "https://www.rust-lang.org/favicon.ico",
        html_root_url = "https://doc.rust-lang.org/time/")]
 #![allow(trivial_numeric_casts)]
+#![allow(unused_imports)]
 #![cfg_attr(test, deny(warnings))]
+#![cfg_attr(test, allow(deprecated))]
+#![cfg_attr(not(feature = "std"), no_std)]
```

Comment:
I'd actually recommend unconditionally having this attribute, and then all the import directives can import from `core` instead of `std` most of the time, hopefully cutting down on a lot of the `#[cfg]` traffic.